sdk/macro/src/lib.rs

```diff
@@ -378,38 +378,19 @@ pub fn pubkeys(input: TokenStream) -> TokenStream {
     TokenStream::from(quote! {#pubkeys})
 }
 
-// Sets padding in structures to zero explicitly.
-// Otherwise padding could be inconsistent across the network and lead to divergence / consensus failures.
-#[proc_macro_derive(CloneZeroed)]
-pub fn derive_clone_zeroed(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+// Asserts that the struct has no padding.
+// Otherwise, padding could be inconsistent across the network and lead to divergence / consensus failures.
+#[proc_macro_derive(NoPadding)]
```

Comment:
I feel like the name of this macro should be `ZeroedPadding` or `PaddingZeroes` (or better!), since that would articulate what it's supposed to do a little bit more. We're not ensuring there's _no padding_, we're ensuring the padding (if it exists) is fact zeroes. What do you think?

Reply:
The macro doesn't add any fields to make padding zeroed so it does actually guarantee no padding. The zeroed padding type is not technically padding